proxy-node-shared/src/main/java/uk/gov/ida/notification/shared/proxy/MetatronProxyFactory.java

```diff
@@ -0,0 +1,23 @@
+package uk.gov.ida.notification.shared.proxy;
+
+import io.dropwizard.client.JerseyClientBuilder;
+import io.dropwizard.client.JerseyClientConfiguration;
+import io.dropwizard.setup.Environment;
+import uk.gov.ida.jerseyclient.ErrorHandlingClient;
+import uk.gov.ida.jerseyclient.JsonResponseProcessor;
+import uk.gov.ida.notification.shared.istio.IstioHeaderStorage;
+import javax.ws.rs.client.Client;
+
+public class MetatronProxyFactory {
```

Comment:
I don't think we need a factory to create a simple class like the Metatron proxy - it only takes two things to build, one of which is a URI. Put all the code in the `buildMetatronProxy` method into the Metatron proxy constructor instead and we can get rid of this class.

Actually we have proxies in a few places now and we should refactor this into a pattern (probably a base Proxy class and a bunch of children)

Reply:
The factory definitely turned out to be overkill but the only thing the proxies really have in common is a Client and a URL and they're just fields; there's no shared functionality or logic so having an underlying base class would also be overkill.

Reply:
👍